test/test_autograd.py

```diff
@@ -2797,6 +2797,49 @@ def run_test(upper, dims):
         for upper, dims in product([True, False], [(3, 3), (5, 3, 3), (4, 3, 2, 2)]):
             run_test(upper, dims)
 
+    @skipIfNoLapack
+    def test_symeig_degenerate(self):
+        # test symeig backward if there are degenerate/repeated eigenvalues
+        dtype = torch.double
+
+        def get_loss(a, mat, P2):
+            # the loss function here is designed so that it depends on the
+            # degenerate eigenvectors, but does not depend on which linear
+            # combination of the degenerate eigenvectors are used.
+
+            # get the orthogonal vector for the eigenvectors
+            P, _ = torch.qr(mat)
+
+            # line up the eigenvalues
+            b = torch.cat((a[:1], a[1:2], a[1:2], a[2:]))
+
+            # construct the matrix
+            diag = torch.diag_embed(b)
+            A = torch.matmul(torch.matmul(P.T, diag), P)
```

Comment:
```suggestion
            A = (P.T * diag.unsqueeze(-2)) @ P
```

Reply:
Thanks, modified it based on the same idea